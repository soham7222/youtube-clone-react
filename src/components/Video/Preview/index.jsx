export {
    Preview as PreviewComponent
} from "./Preview"