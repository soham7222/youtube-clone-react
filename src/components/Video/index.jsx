export { Video as VedioComponent } from "./Video";