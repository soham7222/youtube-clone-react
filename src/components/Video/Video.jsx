import React from 'react'
import { PreviewComponent } from './Preview'
import "./Video.css"

export const Video = (props) => {
    return (
        <div data-testid="video_grid" className="video_grid">{
            props.videos?.map((item, index) => (
                <PreviewComponent key={index} image={item.image.url} title={item.title} channelName={item.channelName} publishTime={item.publishTime} />
            ))}
        </div>
    )
}
