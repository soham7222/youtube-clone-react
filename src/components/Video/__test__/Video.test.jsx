import React from 'react';
import { render } from '@testing-library/react';
import { Video } from '../Video';

describe('Video Component', () => {
    beforeEach(() => {
        jest.spyOn(global.Date, 'now').mockImplementationOnce(() =>
            new Date('2019-05-14T11:01:58.135Z').valueOf()
        );
        jest.spyOn(global.Math, 'random').mockReturnValue(0.123456789);
    });

    afterEach(() => {
        jest.spyOn(global.Math, 'random').mockRestore();
    })

    const videos = [{
        id: 1,
        channelName: "channel 1",
        title: "title 1",
        publishTime: "11:45 PM",
        image: "https://blog.hubspot.com/hubfs/image8-2.jpg"
    }, {
        id: 1,
        channelName: "channel 2",
        title: "title 2",
        publishTime: "11:45 PM",
        image: "https://blog.hubspot.com/hubfs/image8-2.jpg"
    }];

    it('renders correctly', () => {
        const dom = render(<Video videos={videos} />)
        expect(dom).toMatchSnapshot();
    });
})