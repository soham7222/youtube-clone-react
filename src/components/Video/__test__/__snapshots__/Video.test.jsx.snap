// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="video_grid"
        data-testid="video_grid"
      >
        <div
          class="video_preview"
        >
          <div
            class="video_image"
          >
            <img
              class="video_preview_image"
            />
            <div
              class="video_timestamp"
            >
              <span>
                8
                :
                17
              </span>
            </div>
          </div>
          <div
            class="video_info"
          >
            <div
              class="video_channel_logo"
            >
              CH
            </div>
            <div>
              <div
                class="video_info_title"
                title="title 1"
              />
              <div
                class="video_basic_info"
              >
                <div
                  class="video_channel"
                >
                  channel 1
                </div>
                <div
                  class="video_view_time"
                >
                  3
                  k views ● 
                  NaN days ago
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="video_preview"
        >
          <div
            class="video_image"
          >
            <img
              class="video_preview_image"
            />
            <div
              class="video_timestamp"
            >
              <span>
                8
                :
                17
              </span>
            </div>
          </div>
          <div
            class="video_info"
          >
            <div
              class="video_channel_logo"
            >
              CH
            </div>
            <div>
              <div
                class="video_info_title"
                title="title 2"
              />
              <div
                class="video_basic_info"
              >
                <div
                  class="video_channel"
                >
                  channel 2
                </div>
                <div
                  class="video_view_time"
                >
                  3
                  k views ● 
                  NaN days ago
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="video_grid"
      data-testid="video_grid"
    >
      <div
        class="video_preview"
      >
        <div
          class="video_image"
        >
          <img
            class="video_preview_image"
          />
          <div
            class="video_timestamp"
          >
            <span>
              8
              :
              17
            </span>
          </div>
        </div>
        <div
          class="video_info"
        >
          <div
            class="video_channel_logo"
          >
            CH
          </div>
          <div>
            <div
              class="video_info_title"
              title="title 1"
            />
            <div
              class="video_basic_info"
            >
              <div
                class="video_channel"
              >
                channel 1
              </div>
              <div
                class="video_view_time"
              >
                3
                k views ● 
                NaN days ago
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="video_preview"
      >
        <div
          class="video_image"
        >
          <img
            class="video_preview_image"
          />
          <div
            class="video_timestamp"
          >
            <span>
              8
              :
              17
            </span>
          </div>
        </div>
        <div
          class="video_info"
        >
          <div
            class="video_channel_logo"
          >
            CH
          </div>
          <div>
            <div
              class="video_info_title"
              title="title 2"
            />
            <div
              class="video_basic_info"
            >
              <div
                class="video_channel"
              >
                channel 2
              </div>
              <div
                class="video_view_time"
              >
                3
                k views ● 
                NaN days ago
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
